snvphyl-tools/1.8.2/Dockerfile

```diff
@@ -0,0 +1,100 @@
+FROM ubuntu:focal as app
+
+# for easy upgrade later. ARG variables only persist during image build time
+ARG SNVPHYL_VER="1.8.2"
+ARG SAMTOOLSVER="1.9"
+
+# metadata
+LABEL base.image="ubuntu:focal"
+LABEL dockerfile.version="1"
+LABEL software="SNVPhyl-tools"
+LABEL software.version="1.8.2"
+LABEL description="The SNVPhyl-tools are for a pipeline for identifying Single Nucleotide Variants (SNV) within a collection of microbial genomes and constructing a phylogenetic tree."
+LABEL website="https://github.com/phac-nml/snvphyl-tools"
+LABEL license="https://github.com/phac-nml/snvphyl-tools/blob/master/LICENSE"
+LABEL maintainer="Jill Hagey"
+LABEL maintainer.email="jvhagey@gmail.com"
+
+# Install dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    ca-certificates \
+    pkg-config \
+    curl \
+    automake \
+    autoconf \
+    libncurses5-dev \
+    build-essential \
+    zlib1g-dev \
+    libbz2-dev \
+    liblzma-dev \
+    libcurl4-openssl-dev \
+    libexpat1-dev \
+    perl \
+    wget && \
+ rm -rf /var/lib/apt/lists/* && apt-get autoclean
+
+# Install perl package dependencies
```

Comment:
@kapsakcj , is this the best way to install perl packages?

I'm taking note of this of how to install perl dependencies in a docker image.

Reply:
It works, but I've had better success with using `cpanm` instead of `cpan` itself. 

`cpanm` is an alias for CPAN-minus and it much easier to work with, plus it runs faster in most cases. You can install it via apt with `apt install cpanminus`

then install perl packages with `cpanm --notest` which skips testing after installing each perl package and if you need you can add `cpanm --force` to force the install incase it complains for some reason

example here: https://github.com/StaPH-B/docker-builds/blob/dbe5db41991959685b7e0f157f71979f50e528fa/mashtree/1.0.4/Dockerfile#L30

Reply:
oh lordy 26 minutes to build this image? I would definitely try `cpanm --notest <perlpkg> <...>` instead of `cpan`

Reply:
🤦‍♂️ Just realized that you are using `cpanm` on line 38. I'm sorry. Disregard what I said. This is what happens when you read code at midnight